server/chat-plugins/abuse-monitor.ts

```diff
@@ -0,0 +1,560 @@
+	am: 'abusemonitor',
+	abusemonitor: {
+		''() {
+			return this.parse('/join view-abusemonitor-flagged');
+		},
+		async test(target, room, user) {
+			checkAccess(this);
+			const text = target.trim();
+			if (!text) return this.parse(`/help abusemonitor`);
+			this.runBroadcast();
+			const {score, flags} = await PM.query({comment: text});
+			this.sendReplyBox(
+				`Score for "${text}": ${score}<br />` +
+				`Flags: ${flags.join(', ')}`
+			);
+		},
+		toggle(target) {
+			checkAccess(this);
+			const setting = this.meansYes(target);
+			if (settings.disabled === setting) {
+				return this.errorReply(`The abuse monitor is already ${setting ? 'enabled' : 'disabled'}.`);
+			}
+			settings.disabled = setting;
+			saveSettings();
+			this.privateGlobalModAction(`${this.user.name} ${setting ? 'enabled' : 'disabled'} the abuse monitor.`);
+			this.globalModlog('ABUSEMONITOR', null, setting ? 'enable' : 'disable');
+		},
+		threshold(target) {
+			checkAccess(this);
+			if (!target) {
+				return this.sendReply(`The current abuse monitor threshold is ${settings.threshold}.`);
+			}
+			const num = parseInt(target);
+			if (isNaN(num)) {
+				this.errorReply(`Invalid number: ${target}`);
+				return this.parse(`/help abusemonitor`);
+			}
+			if (settings.threshold === num) {
+				return this.errorReply(`The abuse monitor threshold is already ${num}.`);
+			}
+			settings.threshold = num;
+			saveSettings();
+			this.privateGlobalModAction(`${this.user.name} set the abuse monitor trigger threshold to ${num}.`);
+			this.globalModlog('ABUSEMONITOR THRESHOLD', null, `${num}`);
+			this.sendReply(
+				`|html|Remember to use <code>/am respawn</code> to deploy the settings to the child process.`
+			);
+		},
+		resolve(target) {
+			this.checkCan('lock');
+			target = target.toLowerCase().trim().replace(/ /ig, '');
+			if (!target) return this.parse(`/help abusemonitor`);
+			if (!cache[target]?.notified) {
+				return this.popupReply(`That room has not been flagged by the abuse monitor.`);
+			}
+			// we delete the cache because if more stuff happens in it
+			// post punishment, we want to know about it
+			delete cache[target];
+			notifyStaff();
+		},
+		logs(target) {
+			checkAccess(this);
+			const [count, userid] = Utils.splitFirst(target, ',').map(toID);
+			this.parse(`/join view-abusemonitor-logs-${count || '200'}${userid ? `-${userid}` : ""}`);
+		},
+		async respawn(target, room, user) {
+			checkAccess(this);
+			this.sendReply(`Respawning...`);
+			const unspawned = await PM.respawn();
+			this.sendReply(`DONE. ${Chat.count(unspawned, 'processes', 'process')} unspawned.`);
+			this.addGlobalModAction(`${user.name} used /abusemonitor respawn`);
+		},
+		async userclear(target, room, user) {
+			checkAccess(this);
+			const {targetUsername, rest} = this.splitUser(target);
+			const targetId = toID(targetUsername);
+			if (!targetId) return this.parse(`/help abusemonitor`);
+			if (user.lastCommand !== `tm userclear ${targetId}`) {
```

Comment:
more `tm` references

Reply:
Blegh.

Reply:
Though I guess it's technically OK since it's not user-facing.

Reply:
Ok, fixed.